packages/libp2p/src/connection-manager/dial-queue.ts

```diff
@@ -550,15 +588,28 @@ export class DialQueue {
   /**
    * Record the last dial success/failure status of the passed multiaddr
    */
-  private async _updateAddressStatus (peerId: PeerId, multiaddr: Multiaddr, success: boolean): Promise<void> {
+  private async _updateAddressStatus (peerId: PeerId, multiaddr: Multiaddr, success: boolean, peer?: Peer): Promise<void> {
     const addr: Address = {
       multiaddr
     }
 
+    if (peer !== undefined) {
+      const existingAddr = peer.addresses.find(a => a.multiaddr.equals(multiaddr))
+
+      if (existingAddr !== undefined) {
+        addr.lastSuccess = existingAddr.lastSuccess
+        addr.lastFailure = existingAddr.lastFailure
+      }
+    }
+
     if (success) {
       addr.lastSuccess = Date.now()
     } else {
-      addr.lastFailure = Date.now()
+      if (addr.lastFailure !== undefined) {
+        addr.lastFailure *= 2
+      } else {
+        addr.lastFailure = Date.now()
```

Comment:
The `lastFailure` value is a unix timestamp but it's being utilized as a value for ordering the list of multiaddrs, therefore exponentially increasing it so that it will be penalized even further to the bottom of the list seems a valid approach i.e. it could be any unique incremental numerical value as long as it is sequentially increasing to create an order within the list of addrs.